src/uwtools/tests/test_schemas.py

```diff
@@ -456,10 +455,116 @@ def test_schema_rocoto_workflow_cycledef():
 # sfc-climo-gen
 
 
+@fixture
+def sfc_climo_gen_fcstprop():
+    return partial(schema_validator, "sfc-climo-gen", "properties", "sfc_climo_gen", "properties")
+
+
 def test_schema_sfc_climo_gen():
+    d = {
+        "execution": {"executable": "sfc_climo_gen"},
+        "run_dir": "/tmp",
+    }
     errors = schema_validator("sfc-climo-gen", "properties", "sfc_climo_gen")
-    d = test_sfc_climo_gen.config["sfc_climo_gen"]
     # Basic correctness:
     assert not errors(d)
-    # Additional properties are not allowed:
+    # Some top-level keys are required:
+    for key in ("execution", "run_dir"):
+        assert f"'{key}' is a required property" in errors(with_del(d, key))
+    # Additional top-level keys are not allowed:
     assert "Additional properties are not allowed" in errors({**d, "foo": "bar"})
+
+
+def test_schema_sfc_climo_gen_namelist(sfc_climo_gen_fcstprop):
+    base_file = {"base_file": "/some/path"}
+    update_values = {"update_values": {"config": {"var": "val"}}}
+    errors = sfc_climo_gen_fcstprop("namelist")
+    # Just base_file is ok:
+    assert not errors(base_file)
+    # base_file must be a string:
+    assert "88 is not of type 'string'" in errors({"base_file": 88})
+    # Just update_values is ok:
+    assert not errors(update_values)
+    # config is required with update_values:
+    assert "'config' is a required property" in errors({"update_values": {}})
+    # A combination of base_file and update_values is ok:
+    assert not errors({**base_file, **update_values})
+    # At least one is required:
+    assert "is not valid" in errors({})
+
+
+def test_schema_sfc_climo_gen_namelist_update_values(sfc_climo_gen_fcstprop):
+    errors = sfc_climo_gen_fcstprop(
+        "namelist", "properties", "update_values", "properties", "config"
+    )
+    # array, boolean, number, and string values are ok:
+    assert not errors({"array": [1, 2, 3], "bool": True, "int": 88, "float": 3.14, "string": "foo"})
+    # Other types are not, e.g.:
+    assert "None is not of type 'array', 'boolean', 'number', 'string'" in errors({"null": None})
+    # No minimum number of entries is required:
+    assert not errors({})
+
+
+def test_schema_sfc_climo_gen_run_dir(sfc_climo_gen_fcstprop):
+    errors = sfc_climo_gen_fcstprop("run_dir")
+    # Must be a string:
+    assert not errors("/some/path")
+    assert "88 is not of type 'string'" in errors(88)
+
+
+# chgres-cube
```

Comment:
It might be nice to have this block nearer the top, as the existing blocks are all currently in order by component name.

Reply:
Ah, I left them grouped for my work efficiency and never put them back up!

Reply:
Fixed component name sorting and moved all fixtures to the top